Kudu.Services/Diagnostics/ProcessController.cs

```diff
@@ -274,6 +276,35 @@ public HttpResponseMessage MiniDump(int id, int dumpType = 0, string format = nu
             }
         }
 
+        [HttpPost]
+        public HttpResponseMessage TakeCrashDump(int id, bool includeMemory = false)
+        {
+            using (_tracer.Step("ProcessController.TakeCrashDump"))
+            {
+                try
+                {
+                    var process = GetProcessById(id);
+                    var exe = new Executable(ResolveProcDumpPath(), _environment.CrashDumpsPath, TimeSpan.MaxValue);
+                    var memoryFlag = includeMemory ? "-ma" : string.Empty;
+
+                    Task.Run(() => exe.ExecuteReturnExitCode(_tracer, s => _tracer.Trace(s), s => _tracer.TraceError(s), $"-accepteula -t {process.Id} -e -g {memoryFlag}"));
+
+                    return Request.CreateResponse(HttpStatusCode.Accepted);
```

Comment:
We should think about another scenario where process crashes on startup.   This happened pretty often (especially w3wp start up and terminated unexpectedly).   This likely needs to be addressed from Azure code base (not Kudu).